File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/systemui/controlcenter/QQSGrid.java
package com.sevtinge.cemiuiler.module.hook.systemui.controlcenter;

import com.sevtinge.cemiuiler.R;
import com.sevtinge.cemiuiler.module.base.BaseHook;

public class QQSGrid extends BaseHook {
    @Override
    public void init() {
        int cols = mPrefsMap.getInt("system_control_center_old_qs_grid_columns", 2);
        int colsResId = switch (cols) {
            case 3 -> R.integer.quick_quick_settings_num_rows_3;
            case 4 -> R.integer.quick_quick_settings_num_rows_4;
            case 5 -> R.integer.quick_quick_settings_num_rows_5;
            case 6 -> R.integer.quick_quick_settings_num_rows_6;
            case 7 -> R.integer.quick_quick_settings_num_rows_7;
            default -> R.integer.quick_quick_settings_num_rows_5;
        };
        mResHook.setResReplacement("com.android.systemui", "integer", "quick_settings_qqs_count", colsResId);
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/systemui/controlcenter/QQSGrid.kt
package com.sevtinge.cemiuiler.module.hook.systemui.controlcenter

import android.content.res.Configuration
import android.view.ViewGroup

import com.github.kyuubiran.ezxhelper.ClassUtils.loadClass
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.github.kyuubiran.ezxhelper.finders.MethodFinder.`-Static`.methodFinder

import com.sevtinge.cemiuiler.module.base.BaseHook

class QQSGrid : BaseHook() {
    override fun init() {
        val cols = mPrefsMap.getInt("system_control_center_old_qs_grid_columns", 5);
        val colsHorizontal = mPrefsMap.getInt("system_control_center_old_qs_grid_columns_horizontal", 6);

        loadClass("com.android.systemui.qs.MiuiQuickQSPanel").methodFinder().first {
                name == "setMaxTiles" && parameterCount == 1
            }.createHook {
                before {
                    val viewGroup = it.thisObject as ViewGroup
                    val mConfiguration: Configuration = viewGroup.context.resources.configuration
                    if (mConfiguration.orientation == Configuration.ORIENTATION_PORTRAIT) {
                        it.args[0] = cols
                    } else {
                        it.args[0] = colsHorizontal
                    }
                }
            }
    }
}
